examples/mesh_calibration_api.py

```diff
@@ -0,0 +1,267 @@
+                        row_left.append(map_y_l[y - 1, x])
+                        row_left.append(map_x_l[y - 1, x])
+                        row_right.append(map_y_r[y - 1, x])
+                        row_right.append(map_x_r[y - 1, x])
+                    elif x == map_x_l.shape[1]:
+                        row_left.append(map_y_l[y, x - 1])
+                        row_left.append(map_x_l[y, x - 1])
+                        row_right.append(map_y_r[y, x - 1])
+                        row_right.append(map_x_r[y, x - 1])
+                    else:
+                        row_left.append(map_y_l[y, x])
+                        row_left.append(map_x_l[y, x])
+                        row_right.append(map_y_r[y, x])
+                        row_right.append(map_x_r[y, x])
+            if (map_x_l.shape[1] % meshCellSize) % 2 != 0:
+                row_left.append(0)
+                row_left.append(0)
+                row_right.append(0)
+                row_right.append(0)
+
+            mesh_left.append(row_left)
+            mesh_right.append(row_right)
+
+    mesh_left = np.array(mesh_left)
+    mesh_right = np.array(mesh_right)
+    print("Saving mesh to:", output_path)
+    mesh_left.tofile(output_path + "/left_mesh.calib")
+    mesh_right.tofile(output_path + "/right_mesh.calib")
+
+
+# The operations done here seem very CPU-intensive, TODO
+def convert_to_cv2_frame(name, image):
+    global last_rectif_right
+    max_disp = 95
+    if extended:
+        max_disp *= 2
+    if subpixel:
+        max_disp *= 32
+
+    data, w, h = image.getData(), image.getWidth(), image.getHeight()
+    # TODO check image frame type instead of name
+    if name == "rgb_preview":
+        frame = np.array(data).reshape((3, h, w)).transpose(1, 2, 0).astype(np.uint8)
+    elif name == "rgb_video":  # YUV NV12
+        yuv = np.array(data).reshape((h * 3 // 2, w)).astype(np.uint8)
+        frame = cv2.cvtColor(yuv, cv2.COLOR_YUV2BGR_NV12)
+    elif name == "depth":
+        # TODO: this contains FP16 with (lrcheck or extended or subpixel)
+        frame = np.array(data).astype(np.uint8).view(np.uint16).reshape((h, w))
+    else:  # mono streams / single channel
+        frame = np.array(data).reshape((h, w)).astype(np.uint8)
+        if name.startswith("rectified_"):
+            frame = cv2.flip(frame, 1)
+    return frame
+
+
+def test_pipeline():
+    calibData = dai.Device().readCalibration()
+    pipeline, streams = create_stereo_pipeline(calibData, source_camera)  # Also creates mesh files
+
+    print("Creating DepthAI device")
+    with dai.Device(pipeline) as device:
+        # Create a receive queue for each stream
+        q_list = []
+        for s in streams:
+            q = device.getOutputQueue(s, 8, blocking=False)
+            q_list.append(q)
+
+        while True:
+            for q in q_list:
+                name = q.getName()
+                image = q.get()
+                frame = convert_to_cv2_frame(name, image)
```

Comment:
remove `convert_to_cv2_frame` and use frame =  image.getCvFrame().
And for depth add color map for visualization.